controllers/storagecluster/initialization_reconciler_test.go

```diff
@@ -111,8 +117,11 @@ func createUpdateRuntimeObjects(t *testing.T, cp *Platform, r StorageClusterReco
 }
 
 func initStorageClusterResourceCreateUpdateTestWithPlatform(
-	t *testing.T, platform *Platform, runtimeObjs []client.Object) (*testing.T, StorageClusterReconciler, *api.StorageCluster, reconcile.Request) {
+	t *testing.T, platform *Platform, runtimeObjs []client.Object, customSpecs ...api.StorageClusterSpec) (*testing.T, StorageClusterReconciler, *api.StorageCluster, reconcile.Request) {
 	cr := createDefaultStorageCluster()
+	if len(customSpecs) > 0 {
+		_ = mergo.Merge(&cr.Spec, customSpecs[0])
+	}
```

Comment:
handle all values not just the first one.

Reply:
it handles all spec, This is for an optional parameter
func initStorageClusterResourceCreateUpdateTestWithPlatform(
	t *testing.T, platform *Platform, runtimeObjs []client.Object, customSpecs ...api.StorageClusterSpec)

when I pass custom spec it will reived as an array here "customSpecs ...api.StorageClusterSpec"
here we always pass one spec but due to this optional parameter structure it will be received as array-like [spc]
so i am using [0]